pkg/tfbridge/provider.go

```diff
@@ -1760,9 +1762,60 @@ func (p *Provider) Construct(context.Context, *pulumirpc.ConstructRequest) (*pul
 	return nil, status.Error(codes.Unimplemented, "Construct is not yet implemented")
 }
 
-// Call dynamically executes a method in the provider associated with a component resource.
+func (p *Provider) returnTerraformConfig(ctx context.Context) (resource.PropertyMap, error) {
+	resConfig, err := buildTerraformConfig(ctx, p, p.configValues)
+	if err != nil {
+		return nil, fmt.Errorf("error building Terraform config: %v", err)
+	}
+
+	var rawConfig terraform.ResourceConfig
+
+	if cfg, ok := resConfig.(shim.ResourceConfigWithGetterForSdkV2); ok {
+		rawConfig = cfg.GetTFConfig()
+	}
+
+	if !rawConfig.CtyValue.IsWhollyKnown() {
+		return nil, fmt.Errorf("cannot use terraformConfig that contains unknown cty values")
+	}
+
+	configJSONMessage, err := valueshim.FromHCtyValue(rawConfig.CtyValue).Marshal()
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling into raw JSON message: %v", err)
+	}
+
+	jsonConfigMap := map[string]any{}
+	err = json.Unmarshal(configJSONMessage, &jsonConfigMap)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling JSON: %v", err)
+	}
+
+	return resource.NewPropertyMapFromMap(jsonConfigMap), nil
+}
+
+// Call dynamically executes a method in the provider.
 func (p *Provider) Call(ctx context.Context, req *pulumirpc.CallRequest) (*pulumirpc.CallResponse, error) {
```

Comment:
Looks like Call implementation is exactly the same between PF and SDKv2 so maybe we could find a way to share the code to not double-state it?